### PR TITLE
Support Nodejs v18

### PR DESCRIPTION
**Pull request by fbaldo31**

### CODE REVIEW

**Comment by inquinity** [comment]
Why node 18? At think point, we should go straight to node 20.

**Comment by fbaldo31** [comment]
I agree the target should be Node 20, but I prefer to go step by step:
I didn't tested with Node 20 yet but I am afraid to meet issues with dependencies.
As Node 18 is still maintained untill 2025-04-30 it can be a first step before going to current LTS (20)

File: packages/core/src/main/__test__/prometheus-handler.test.ts [hunk 1]
```diff
@@ -186,7 +186,7 @@ describe("PrometheusHandler", () => {
       [2, 2],
       [2, 3],
     ])("should pick the first provider of %d success(es) between %d failure(s)", async (successes, failures) => {
-      const beforeSuccesses = Math.floor(successes / 2);
+      const beforeSuccesses = Math.floor(successes / 4);
```
Comment (gdubcs) [inline packages/core/src/main/__test__/prometheus-handler.test.ts:189]: Don't know what that does ! :-D

**Review by gdubcs — APPROVED** [review]

**Comment by colintufts** [comment]
> I agree the target should be Node 20, but I prefer to go step by step: I didn't tested with Node 20 yet but I am afraid to meet issues with dependencies. As Node 18 is still maintained untill 2025-04-30 it can be a first step before going to current LTS (20)

I agree that it is a good starting point. I might have missed something with the prometheus handler test, would you mind explaining why we're now only looking for a quarter of the tests to pass? Does Node v18 require the tests to complete faster?

**Comment by fbaldo31** [comment]
> > I agree the target should be Node 20, but I prefer to go step by step: I didn't tested with Node 20 yet but I am afraid to meet issues with dependencies. As Node 18 is still maintained untill 2025-04-30 it can be a first step before going to current LTS (20)
> 
> I agree that it is a good starting point. I might have missed something with the prometheus handler test, would you mind explaining why we're now only looking for a quarter of the tests to pass? Does Node v18 require the tests to complete faster?

For PrometheusHandler test I am not sure why previous setup fails on Node 18. Maybe you can find a better way to handle this.

**Comment by colintufts** [comment]
> > > I agree the target should be Node 20, but I prefer to go step by step: I didn't tested with Node 20 yet but I am afraid to meet issues with dependencies. As Node 18 is still maintained untill 2025-04-30 it can be a first step before going to current LTS (20)
> > 
> > 
> > I agree that it is a good starting point. I might have missed something with the prometheus handler test, would you mind explaining why we're now only looking for a quarter of the tests to pass? Does Node v18 require the tests to complete faster?
> 
> For PrometheusHandler test I am not sure why previous setup fails on Node 18. Maybe you can find a better way to handle this.

No problem, I was more curious than anything.

**Pull request merged**